File: app/src/main/java/com/github/pokatomnik/frontendconf2023/examples/ex08sideeffects/ex08SideEffects.kt
package com.github.pokatomnik.frontendconf2023.examples.ex08sideeffects

import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect

fun toastStub(message: String) {
    // Показываем здесь сообщение
}

@Composable
fun useToast(message: String) {
    LaunchedEffect(message) {
        toastStub(message)
    }
}

File: app/src/main/java/com/github/pokatomnik/frontendconf2023/examples/ex08sideeffects/ex08SideEffects.ts
const toastStub = (message: string) => {
  // Показываем здесь сообщение
};

export const useToast = (props: {
  readonly message: string
}) => {
  const { message } = props;
  React.useEffect(() => {
    toastStub(message);
  }, [message]);
};
